test/types/index.ts

```diff
@@ -2,12 +2,15 @@
 // This file will be passed to the TypeScript CLI to verify our typings compile
 
 import * as fastify from '../../fastify'
-import * as cors from 'cors'
 import * as http from 'http';
 import * as http2 from 'http2';
 import { readFileSync } from 'fs'
 import { createReadStream, readFile } from 'fs'
 
+// were importing cors using require, which causes it to be an `any`. This is done because `cors` exports
+// itself as an express.RequestHandler which is not compatible with the fastify TypeScript types
```

Comment:
Could we cast it to a compatible middleware interface that we define in fastify?

Reply:
we can cast to whatever we want, but it does not prove anything.
If we cast, all we prove is that the cast is a correct cast.
But does not prove that the lib works correctly with middleware.

If you wanted to cast you can test by casting any function, there would be no benefit to importing and casting `cors`

Reply:
@evanshortiss what do you think?

Reply:
@tikotzky @mcollina I don't think there's much benefit to casting either. Could there be some benefit to changing the type signature of `fastify.use` to accept `express.RequestHandler` instances?

Reply:
@evanshortiss they are not the same. Some middlewares are compatible, but not all. Specifically, the types of `cors` are wrong as that should be a standard http response from core, not an Express one.

Reply:
Aah, you're correct. Maybe something to revisit in a separate issue/pr?